File: Source/WickedSick.Fayde.Client.Engine/Javascript/Controls/RowDefinition.js
/// <reference path="../Runtime/Nullstone.js" />
/// <reference path="../Core/DependencyObject.js"/>
/// <reference path="GridLength.js"/>
/// CODE

(function (namespace) {
    var RowDefinition = Nullstone.Create("RowDefinition", DependencyObject);

    //#region Properties

    RowDefinition.HeightProperty = DependencyProperty.RegisterCore("Height", function () { return namespace.GridLength; }, RowDefinition, new namespace.GridLength(1.0, namespace.GridUnitType.Star));
    RowDefinition.MaxHeightProperty = DependencyProperty.RegisterCore("MaxHeight", function () { return Number; }, RowDefinition, Number.POSITIVE_INFINITY);
    RowDefinition.MinHeightProperty = DependencyProperty.RegisterCore("MinHeight", function () { return Number; }, RowDefinition, 0.0);
    RowDefinition.ActualHeightProperty = DependencyProperty.RegisterReadOnlyCore("ActualHeight", function () { return Number; }, RowDefinition, 0.0);

    Nullstone.AutoProperties(RowDefinition, [
        RowDefinition.HeightProperty,
        RowDefinition.MaxHeightProperty,
        RowDefinition.MinHeightProperty,
        RowDefinition.ActualHeightProperty
    ]);

    //#endregion

    namespace.RowDefinition = Nullstone.FinishCreate(RowDefinition);
})(Nullstone.Namespace("Fayde.Controls"));

File: Source/Fayde.Server/Controls/RowDefinition.cs
using Fayde.Core;
using Fayde.Xaml.Metadata;

namespace Fayde.Controls
{
    public class RowDefinition : DependencyObject
    {
        public static readonly PropertyDescription HeightProperty = PropertyDescription.Register("Height", typeof(GridLength), typeof(RowDefinition));
        public static readonly PropertyDescription MaxHeightProperty = PropertyDescription.Register("MaxHeight", typeof(Double), typeof(RowDefinition));
        public static readonly PropertyDescription MinHeightProperty = PropertyDescription.Register("MinHeight", typeof(Double), typeof(RowDefinition));
    }
}